eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -755,70 +756,56 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
-      // channels are publicly announced if both parties want it (defined as feature bit)
-      d.shortIds.real match {
-        case RealScidStatus.Final(realScid) =>
-          // we are aware that the channel has reached enough confirmations
-          // we already had sent our announcement_signatures but we don't store them so we need to recompute it
-          val localAnnSigs = Helpers.makeAnnouncementSignatures(nodeParams, d.commitments.params, d.commitments.latest.remoteFundingPubKey, realScid)
+      // Channels are publicly announced if both parties want it: we ignore this message if we don't want to announce the channel.
+      d.shortIds.real_opt match {
+        case Some(realScid) if remoteAnnSigs.shortChannelId == realScid =>
           d.channelAnnouncement match {
             case None =>
-              require(localAnnSigs.shortChannelId == remoteAnnSigs.shortChannelId, s"shortChannelId mismatch: local=${localAnnSigs.shortChannelId} remote=${remoteAnnSigs.shortChannelId}")
-              log.info(s"announcing channelId=${d.channelId} on the network with shortId=${localAnnSigs.shortChannelId}")
+              log.info("announcing channelId={} on the network with shortChannelId={}", d.channelId, realScid)
+              // We already sent our announcement_signatures but we don't store them so we need to recompute them.
+              val localAnnSigs = Helpers.makeAnnouncementSignatures(nodeParams, d.commitments.params, d.commitments.latest.remoteFundingPubKey, realScid)
               val fundingPubKey = keyManager.fundingPublicKey(d.commitments.params.localParams.fundingKeyPath, fundingTxIndex = 0) // TODO: public announcements are not yet supported with splices
               val channelAnn = Announcements.makeChannelAnnouncement(nodeParams.chainHash, localAnnSigs.shortChannelId, nodeParams.nodeId, d.commitments.params.remoteParams.nodeId, fundingPubKey.publicKey, d.commitments.latest.remoteFundingPubKey, localAnnSigs.nodeSignature, remoteAnnSigs.nodeSignature, localAnnSigs.bitcoinSignature, remoteAnnSigs.bitcoinSignature)
               if (!Announcements.checkSigs(channelAnn)) {
                 handleLocalError(InvalidAnnouncementSignatures(d.channelId, remoteAnnSigs), d, Some(remoteAnnSigs))
               } else {
-                // we generate a new channel_update because the scid used may change if we were previously using an alias
+                // We generate a new channel_update because we can now use the scid of the announced funding transaction.
                 val scidForChannelUpdate = Helpers.scidForChannelUpdate(Some(channelAnn), d.shortIds.localAlias)
                 val channelUpdate = Helpers.makeChannelUpdate(nodeParams, remoteNodeId, scidForChannelUpdate, d.commitments, d.channelUpdate.relayFees)
-                // we use goto() instead of stay() because we want to fire transitions
+                announcementSigsSent += remoteAnnSigs.shortChannelId
```

Comment:
Why do we increment `announcementSigsSent` here? We are not sending our `localAnnSigs` AFAICT?